package-res/resources/web/vizapi/VizController.js

```diff
@@ -350,7 +350,7 @@ define([
         chartExists();
       } else {
         var chartDiv = this.visualPanelElement;
-        vizTypes.createByTypeAync(visualization, chartDiv, function(viz) {
+        vizTypes.createByTypeAync(visualization, chartDiv).then(function(viz) {
```

Comment:
Well spotted! This method is temporary and wasn't even unit-tested :-P I think it is only being used by the VizController, I'll correct that.